openverse_catalog/dags/providers/provider_api_scripts/nypl.py

```diff
@@ -155,31 +160,54 @@ def _get_creators(creatorinfo):
         creator = None
 
     if creator is None:
-        logger.warning("No primary creator found")
+        logger.debug("No primary creator found")
 
     return creator
 
 
-def _get_image_url(images, image_url_dimensions=None):
-    if image_url_dimensions is None:
-        image_url_dimensions = IMAGE_URL_DIMENSIONS
-    image_type = {
-        parse_qs(urlparse(img.get("$")).query)["t"][0]: img.get("$") for img in images
+def _get_filetype(description: str):
+    """
+    Extracts the filetype from a description string like:
+    "Cropped .jpeg (1600 pixels on the long side)"
+    :param description: the description string
+    :return:  jpeg | gif
```

Comment:
That's so intense that there isn't just a file name extension in a url somewhere. [Right?](https://digitalcollections.nypl.org/items/5d399174-5bd8-4e97-95ad-a65c8204d892) How did you decide to limit the regex to jpeg and gif? (e.g. and not jpg or png) Also, how is the file type data used in the front end?

Reply:
Yes, such URLs really pose a problem when adding data. This happens more often for audio files than for audio.

But NYPL is actually not that bad: it only has `jpeg`s and `gif`s (and very huge original `tiff`s that we do not use). `gif` is usually used for a very small version of the image, so it's actually mostly `jpeg`.

File type data is used for filtering the images by extension, both in the API and in the front end. File type also should be displayed on the individual result pages.